python-checks/src/main/java/org/sonar/python/checks/LoopExecutingAtMostOnceCheck.java

```diff
@@ -93,6 +93,11 @@ private static void checkLoop(CfgBranchingBlock loopBlock, SubscriptionContext c
     jumps.forEach(j -> issue.secondary(j, null));
   }
 
+  private static boolean jumpOfCurrentLoop(CfgBlock block, Tree loop) {
+    Tree jumpStatement = block.elements().get(block.elements().size() - 1);
+    return TreeUtils.firstAncestorOfKind(jumpStatement, Kind.WHILE_STMT, Kind.FOR_STMT) == loop;
```

Comment:
I am wondering why we need to get back to syntax level for this method : shouldn't the CFG already answer this question about what loops "holds" the break/continue statement ? in other words : couldn't we check that the loop condition block is somewhat the successor of the block containing this break/continue ? 
It seems to me we are reimplementing some logic already solved by CFG (maybe because we lack an API ? )

(wondering if something like, successors.filter(CFGBranchingBlock).anyMatch(branchingTree == loop) could be enough ? )

Reply:
The successor of a block having `break` as terminator is going to be the block following the loop (not the loop condition block), so I think it wouldn't be enough.

Inside the CFG builder we know what is going to be the "target" for break, but I think we would need to store extra information to know to which loop it belongs (and consequently having an API to access this) and I'm not sure we should that in the scope of this PR. What do you think?